contracts/oracles/composite/CompositeMultiOracle.sol

```diff
@@ -31,74 +35,119 @@ contract CompositeMultiOracle is IOracle, AccessControl {
         }
     }
 
-    /// @dev Set or reset an price path
-    function setPath(bytes6 base, bytes6 quote, bytes6[] memory path)
-        external auth
-    {
-        bytes6[] memory reverse = new bytes6[](path.length);
+    /// @notice Set or reset an price path and its reverse path
+    /// @param base Id of base token
+    /// @param quote Id of quote token
+    /// @param path Path from base to quote
+    function setPath(
+        bytes6 base,
+        bytes6 quote,
+        bytes6[] memory path
+    ) external auth {
+        uint256 pathLength = path.length;
+        bytes6[] memory reverse = new bytes6[](pathLength);
         bytes6 base_ = base;
-        for (uint256 p = 0; p < path.length; p++) {
-            require (sources[base_][path[p]] != IOracle(address(0)), "Source not found");
-            base_ = path[p];
-            reverse[path.length - (p + 1)] = base_;
+        unchecked {
+            for (uint256 p = 0; p < pathLength; ++p) {
+                require(sources[base_][path[p]] != IOracle(address(0)), "Source not found");
+                base_ = path[p];
+                reverse[pathLength - (p + 1)] = base_;
+            }
         }
         paths[base][quote] = path;
         paths[quote][base] = reverse;
         emit PathSet(base, quote, path);
         emit PathSet(quote, base, path);
     }
 
-    /// @dev Convert amountBase base into quote at the latest oracle price, through a path is exists.
-    function peek(bytes32 base, bytes32 quote, uint256 amountBase)
-        external view virtual override
-        returns (uint256 amountQuote, uint256 updateTime)
-    {
+    /// @notice Convert amountBase base into quote at the latest oracle price, through a path is exists.
+    /// @param base Id of base token
+    /// @param quote Id of quote token
+    /// @param amountBase Amount of base to convert to quote
+    /// @return amountQuote Amount of quote token converted from base
+    function peek(
+        bytes32 base,
+        bytes32 quote,
+        uint256 amountBase
+    ) external view virtual override returns (uint256 amountQuote, uint256 updateTime) {
```

Comment:
I like to break the `returns` into its own line, but that's just me.

Reply:
I agree it looks nicer but Prettier keeps moving it back up after I change it so until i figure out how to change the prettier rule, I will leave it